### PR TITLE
Add prow env for beta integration test

### LINKED ISSUE #5734: V1 pull-tekton-pipeline-beta-integration-tests
### Feature request

Prow env needs to be updated in the pipeline repo.

As v1 got introduced in #5579  , the feature flag gets a new valid value of `beta`. Therefore a new CI job `pull-tekton-pipeline-beta-integration-tests` to test features that are marked as beta in the v1 API will be needed. 

Since during the gap between v1 gets `served` in 0.42 release and when it gets stored in the next release (0.43 expected), the v1beta1 api is still served and needs to be tested.

### Use case
The beta features will be on by default in v1beta1 testing (which is now `pull-tekton-pipeline-integration-tests`) and `beta` features in v1 would be tested with a feature-flag of `beta`.

### PR DESCRIPTION
**Pull request by JeromeJu**
This commit adds the prow env for beta integration test

# Changes
/kind feature
fixes #5734 

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [n/a] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Adds pull-tekton-pipelines-beta-integration-test
```

### CODE REVIEW

**Comment by JeromeJu** [comment]
cc @dibyom

**Review by dibyom — APPROVED** [review]
This looks good but I think we need #5579 to be merged before this runs any tests

**Review by chitrangpatel — COMMENTED** [review]
Do you think we also need to update the e2e-tests.sh or e2e-common.sh to pick up the new environment?

**Comment by JeromeJu** [comment]
> 

Thanks Chitrang, could you help elaborate what do you mean by `pick up the new environment`?

**Comment by chitrangpatel** [comment]
> Thanks Chitrang, could you help elaborate more about what do you mean by `pick up the new environment`?
> If it's about what's under the `beta` tests, we might want to wait for #5579 and its follow-up as in #5706. As right now there're still features that have not been updated as beta yet eg. [#5540 (comment)](https://github.com/tektoncd/pipeline/pull/5540#discussion_r998301542)

Sorry, I was under the impression that some script has to trigger that environment (ie. set those fields). I think the prow cluster does that for us so might point is probably not valid.

**Comment by JeromeJu** [comment]
/hold
Needs #5579

**Review by chitrangpatel — APPROVED** [review]

File: test/README.md
Comment (XinruZhang) [inline test/README.md:33]: It seems like the link does not match the text 🤔

Comment (JeromeJu) [reply test/README.md:33]: Thanks for the catch, will change to permanent link

Comment (JeromeJu) [reply test/README.md:33]: Actually for this it seems we have to wait for https://github.com/tektoncd/plumbing/pull/1256

**Comment by dibyom** [comment]
I think this can now be merged? 
cc @abayer @XinruZhang

File: test/README.md
Comment (XinruZhang) [reply test/README.md:33]: Ahh, either we need to remove this in the PR and create a TODO issue to catch it as a followup, or we need to stop using permanent link here, or we hold this PR untile that PR gets merged.

wdyt?

**Comment by XinruZhang** [comment]
We may need to hold this until https://github.com/tektoncd/plumbing/pull/1256 gets merged (and stop using the permenant link when referring to the new unit test), see https://github.com/tektoncd/pipeline/pull/5737#discussion_r1028532106

/hold

**Comment by chitrangpatel** [comment]
> We may need to hold this until [tektoncd/plumbing#1256](https://github.com/tektoncd/plumbing/pull/1256) gets merged (and stop using the permenant link when referring to the new integration test), see [#5737 (comment)](https://github.com/tektoncd/pipeline/pull/5737#discussion_r1028532106)
> 
> /hold

I think it should be the other way around. Before we merge the plumbing PR, we should have this in place. Otherwise, if jobs are triggered, the prow config will not have anything to point to in the pipeline repo until this is merged.

**Comment by XinruZhang** [comment]
> > We may need to hold this until [tektoncd/plumbing#1256](https://github.com/tektoncd/plumbing/pull/1256) gets merged (and stop using the permenant link when referring to the new integration test), see [#5737 (comment)](https://github.com/tektoncd/pipeline/pull/5737#discussion_r1028532106)
> > /hold
> 
> I think it should be the other way around. Before we merge the plumbing PR, we should have this in place. Otherwise, if jobs are triggered, the prow config will not have anything to point to in the pipeline repo until this is merged.

I see, sounds good, then at least let's add the link referring to the new integration test after this PR and https://github.com/tektoncd/plumbing/pull/1256 both merged. (Simply because the current link is wrong)

**Comment by dibyom** [comment]
Agree with @chitrangpatel also agree with fixing the broken link :D

File: test/README.md
Comment (JeromeJu) [reply test/README.md:33]: Thanks, updated

**Review by XinruZhang — APPROVED** [review]
Thanks @JeromeJu !

**Comment by chitrangpatel** [comment]
@JeromeJu, @XinruZhang can we cancel the hold here?

**Comment by JeromeJu** [comment]
> @JeromeJu, @XinruZhang can we cancel the hold here?

/unhold

**Pull request merged**
